src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,202 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class XdocsUrlTest {
+
+    public static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    public static final String NODE_NAME = "a";
+
+    public static final String SUFFIX_JAVA = ".java";
+
+    public static final String SUFFIX_CHECK_JAVA = "Check.java";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    private static String getRootPath() {
+        return System.getProperty("user.dir");
+    }
+
+    private static String getXmlPath() {
+        return "src/xdocs/checks.xml";
+    }
+
+    private Map<String, String[]> getXdocsMap() {
+        final String rootPath = getRootPath();
+        final File checks = new File(rootPath + "/" + PACKAGE_NAME);
+        final String[] checksSubDirNames = checks.list((rootdir, pathname) -> {
+            return !pathname.endsWith(SUFFIX_JAVA);
+        });
+        final Map<String, String[]> checksNamesMap = new HashMap<>();
+        if (checksSubDirNames != null) {
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(rootPath + "/" + PACKAGE_NAME + "/" + dirName);
+                final String[] fileNames = dirFile.list((rootdir, pathname) -> {
+                    return pathname.endsWith(SUFFIX_CHECK_JAVA);
+                });
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list((rootdir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            });
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest myhandler = new CheckTest(NODE_NAME);
+        final InputStream input = Files.newInputStream(
```

Comment:
I try my best but not find a way to solve CloseResource violation in pmd. If I add try-catch to fix violation in pmd, I will get another violation that Checkstyle only allow failure in try-catch when this is Test class.

Reply:
done